R/check_code.R

```diff
@@ -1,55 +1,72 @@
+#' Check code against a solution.
 #'
 #' Checks the code expression or the code result against a solution.
 #'
-#' \code{check_code()} compares student code to a solution (i.e. model code) and
+#' `check_code()` compares student code to a solution (i.e. model code) and
 #' describes the first way that the student code differs. If the student code
-#' exactly matches the solution, \code{check_code()} returns a customizable
-#' success message (\code{correct}). If the student code does not match the
-#' solution, a customizable incorrect message (\code{incorrect}) can also be
-#' provided.
+#' exactly matches the solution, `check_code()` returns a customizable success
+#' message (`correct`). If the student code does not match the solution, a
+#' customizable incorrect message (`incorrect`) can also be provided.
 #'
-#' \code{check_code()} provides a *strict* check in that the student code must
+#' `check_code()` provides a *strict* check in that the student code must
 #' exactly match the solution. It is not enough for the student code to be
 #' equivalent to the solution code (e.g. to return the same result as the
 #' solution).
 #'
-#' You can provide solution code for \code{check_code()} to use in two ways:
+#' You can provide solution code for `check_code()` to use in two ways:
 #'
 #' 1. Pass code as a character string or a quoted expression to the solution
-#' argument of \code{check_code()}
+#' argument of `check_code()`
 #'
 #' 2. Make a "-solution" code chunk for the exercise to be checked in a learnr
-#' document. There is no need to supply a solution argument for
-#' \code{check_code()} if you call it from the "-check" chunk of the same
-#' exercise. Likewise, there is no need to supply a student submitted code
-#' argument when you call \code{check_code()} from a learnr document (learnr
-#' will provide the code that the student submits when it runs
-#' \code{check_code()}.
+#' document. There is no need to supply a solution argument for `check_code()`
+#' if you call it from the "-check" chunk of the same exercise. Likewise, there
+#' is no need to supply a student submitted code argument when you call
+#' `check_code()` from a learnr document (learnr will provide the code that the
+#' student submits when it runs `check_code()`.
 #'
 #' For best results, name all arguments provided in the solution code.
 #'
-#' @template correct
-#' @template incorrect
-#' @template grader_args
-#' @template learnr_args
-#' @template glue_correct
-#' @template glue_incorrect
+#' @param correct A character string to display if the student answer matches a
+#'   known correct answer.
+#'
+#' @param incorrect A character string to display if the student answer matches
+#'   a known incorrect answer.
+#'
+#' @param grader_args A list of parameters passed to `grader` functions
+#'   (provided by [grade_learnr()]). This contains:
+#'
+#'   * `user_quo`: Quoted R code submitted by the user. For example
+#'   `[rlang::quo](1)`
```

Comment:
`[rlang::quo(1)]` ?

Might have to use `\code{\link[rlang:quo]{rlang::quo(1)}}`

Reply:
is that correct? to mix both LaTeX and markdown?

Reply:
The roxygen markdown has limited support for everything you can do in a `.Rd` file.  If you run out of options in roxygen md, then it's legitimate to use the latex-like notation.  However, in general I try to avoid this, since it usually means there is another, simpler way of doing things.

Reply:
I fixed this in d171ed3c671b6d9f1c8217bb45da6bb31a729f6d